### PR TITLE
[SPARK-39836][SQL] Simplify V2ExpressionBuilder by extract common method.

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, `V2ExpressionBuilder` have a lot of similar code, we can extract them as one common method.

We can simplify the implement with the common method.

### Why are the changes needed?
Simplify `V2ExpressionBuilder` by extract common method.

### Does this PR introduce _any_ user-facing change?
'No'.
Just update inner implementation.

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by beliefer** [comment]
ping @huaxingao cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala [hunk 1]
```diff
@@ -92,42 +92,17 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       generateExpression(child).map(v => new V2Cast(v, dataType))
     case Abs(child, true) => generateExpression(child)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala:93]: I think unary expression can use the new `generateExpressionWithName` as here
```
case Abs(child, true) => generateExpressionWithName("ABS", Seq(child))
```

**Review by huaxingao — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
The failed tests are unrelated to DS v2, thanks, merging to master!

**Comment by beliefer** [comment]
@cloud-fan @huaxingao Thank you!

**Pull request closed without merging**